py/htm/encoders/date.py

```diff
@@ -249,12 +250,18 @@ def encode(self, inp, output=None):
     if self.seasonEncoder is not None:
       # Number the days starting at zero, intead of 1 like the datetime does.
       dayOfYear = timetuple.tm_yday - 1
+      assert(dayOfYear >= 0)
       # dayOfYear -= self.seasonEncoder.parameters.radius / 2. # Round towards the middle of the season.
       sdrs.append( self.seasonEncoder.encode(dayOfYear) )
 
     if self.dayOfWeekEncoder is not None:
-      dayOfWeek = timetuple.tm_wday + (timeOfDay) / 24.0
-      dayOfWeek -= .5 # Round towards noon, not midnight
+      hrs_ = float(timeOfDay) / 24.0 # add hours as decimal value in extension to day  
+      dayOfWeek = timetuple.tm_wday + hrs_
+      dayOfWeek -= .5 # Round towards noon, not midnight, this means similarity of representations changes at midnights, not noon.
+      # handle underflow: on Mon before noon -> move to Sun
+      if dayOfWeek < 0:
+        dayOfWeek += 7
```

Comment:
A came with another fix, that reverts the `bias`, so days are compared against noon. 
- this is desirable so that dates are "similar" during day-time [morning, eve], and change representation at midnight.
- compatible behavior with Numenta's

The fix was to handle the underflow by +7 (aka comparing to prev Sunday)

Reply:
`Round towards noon, not midnight, this means similarity of representations changes at midnights, not noon.`

Actually, the similarity changes every hour throughout the day.  The effect you're describing here only happens in the unrealistic case of having a single bit per day.

Reply:
ok, true. "In the too-few-bits scenario, when the 2 representations don't overlap"